buildSrc/src/integTest/groovy/org/elasticsearch/gradle/TestClustersPluginFuncTest.groovy

```diff
@@ -75,6 +75,31 @@ class TestClustersPluginFuncTest extends AbstractGradleFuncTest {
         assertNoCustomDistro('myCluster')
     }
 
+    def "can declare test cluster in lazy evaluated task configuration block"() {
```

Comment:
The original problem we wanna fix here, is that we cannot port the RestTestTasks to use the task avoidance api as relying on the afterEvaluate for finalizing the test cluster, as with task avoidance api the materialisation of a RestTestTask (which creates a matching TestCluster instance) happens to late when afterEvaluate has been already triggered. I think ideally we would configure those clusters that are tight to those test tasks in a 1:1 manner more in the context of this test task. something like:

```
tasks.register('someRestTest', RestTestTask) {
   cluster {
      testDistribution = 'DEFAULT'
      setting 'xpack.ml.enabled', 'false'
      setting 'xpack.watcher.enabled', 'false'
    }
}
```   

IMO this makes things also more readable, especially for projects with multiple rest test tasks.

Reply:
Would doing `.all()` instead of `.configureEach()` trigger those tasks to be created? I suppose it would, huh. I guess there's no API to say "do this thing for each "registered" task of this type".

Reply:
I think for now this is the only way to go. Calling it good.